domain/secret/watcher_test.go

```diff
@@ -279,6 +279,57 @@ func (s *watcherSuite) TestWatchObsoleteForUnitsOwned(c *gc.C) {
 	wCSingleUnit.AssertNoChange()
 }
 
+func (s *watcherSuite) TestWatchObsoleteForUserSecrets(c *gc.C) {
+	ctx := context.Background()
+	svc, st := s.setupServiceAndState(c)
+
+	data := coresecrets.SecretData{"foo": "bar", "hello": "world"}
+	uri1 := coresecrets.NewURI()
+	uri2 := coresecrets.NewURI()
+	c.Logf("uri1: %v, uri2: %v", uri1, uri2)
+
+	err := st.CreateUserSecret(ctx, 1, uri1, secret.UpsertSecretParams{
+		Data: data,
+	})
+	c.Assert(err, jc.ErrorIsNil)
+	err = st.CreateUserSecret(ctx, 1, uri2, secret.UpsertSecretParams{
+		Data:      data,
+		AutoPrune: ptr(true),
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	w, err := svc.WatchObsoleteUserSecretsToPrune(ctx)
+	c.Assert(err, gc.IsNil)
+	c.Assert(w, gc.NotNil)
+	defer workertest.CleanKill(c, w)
+
+	wc := watchertest.NewNotifyWatcherC(c, w)
+
+	// Wait for the initial changes.
+	wc.AssertOneChange()
+
+	// create revision 2, and obsolete revision 1.
+	createNewRevision(c, st, uri1)
+	wc.AssertNoChange()
+	createNewRevision(c, st, uri2)
+	wc.AssertAtLeastOneChange()
```

Comment:
Why at least 1? Shouldn't there only be 1 change

Reply:
The single-column ChangeLog watcher implementation has been deleted(Before, we can only watch the `auto-prune` change but now updates on any column will insert a change log record).
The updateSecret state method is an upsert operation that will always update the `secret_metadata."update-time"` field.
Therefore, whenever a new revision is created, we will trigger an auto-prune change event.
Here, we will get two events, unfortunately.

Reply:
We know there will be 2 changes right? Since there's 2 tables as you say. Can we add a helper to test for the expected number of changes?